File: leetcode/php/[1920]Build-Array-from-Permutation.php
<?php
class Solution {

  /**
   * @param Integer[] $nums
   * @return Integer[]
   */
  function buildArray($nums) {
      $arr = [];
      for($i = 0; $i <= count($nums)-1; $i++) {
          $arr[] = $nums[$nums[$i]];
      }
      return $arr;
  }
}

File: leetcode/csharp/[1920]Build-Array-from-Permutation.cs
public class Solution {
    public int[] BuildArray(int[] nums) {
        int[] arr = {};
        for(int i = 0; i <= nums.Length -1 ; i++) {
            // 配列の要素数を増やす
            Array.Resize(ref arr, arr.Length + 1);
            // 配列に要素を追加する
            arr[arr.Length -1] = nums[nums[i]];
        }
        return arr;

    }
}